### ISSUE TITLE
Refactor pipeline config

### ISSUE THREAD

**Issue by gitttt-1234** [talmolab/sleap-nn#66]
1. **_Rearrange preprocessing config_**.
The preprocessing related parameters like scale and is_rgb are defined outside the `preprocessing` section of the config, and should be placed inside the `preprocessing` section.
2. **_Combine `train` and `val` preprocess config_**.
There are two section in the config file for train and validation related parameters. However, the preprocessing config is same across both the train and validation set. The `train` and `val` sections in the `data_config` should be combined to remove redundancy.
3. **Explicit Parameters** especially for DataLoader and model checkpoint
4. Combine devices in config - currently there are 2 parameters: device and trainer_accelerator
5. Remove pipeline and head names from config. The model type should be inferred from head_config keys. Refer: [SLEAP](https://github.com/talmolab/sleap/blob/38a5ca785d02bae74e09b4102635d6711f096c46/tests/data/models/minimal_robot.UNet.single_instance/initial_config.json#L41). Remove redundant `backbone_config` key from config.
